CMakePresets.json

```diff
@@ -0,0 +1,97 @@
+{
+    "version": 3,
+    "configurePresets": [
+        {
+            "name": "common",
+            "hidden": true,
+            "generator": "Ninja",
+            "binaryDir": "build/${presetName}",
```

Comment:
What does this control? Only the literal binary directory? Should we control it like this?

Reply:
Yes I think that really puts the build directly somewhere in case it is not overwritten. I guess it does not harm but also I don't think we really need this and want to control this

Reply:
I kind of goes against where I usually put it 😅 so I wouldn't mind not controlling it. 

OTOH: this can be overridden with user preset, right?

Reply:
yeah exactly I think it just acts as a default if the user does not provide one. I will drop it for now as I don't think we want to suggest something here. I think I just copied it from somewhere